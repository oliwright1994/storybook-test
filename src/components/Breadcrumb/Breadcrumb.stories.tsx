import createStory from '@bright/storybook/createStory'
import React from 'react'

import Breadcrumb from './Breadcrumb'

createStory('COMPONENTS | Breadcrumb', [
  {
    label: 'Breadcrumb',
    jsx: () => {
      return (
        <Breadcrumb>
          <Breadcrumb.Item href="/">Home</Breadcrumb.Item>
          <Breadcrumb.Item href="/about">About</Breadcrumb.Item>
          <Breadcrumb.Item href="/contact" isCurrentPage>
            Contact
          </Breadcrumb.Item>
        </Breadcrumb>
      )
    },
  },
])
