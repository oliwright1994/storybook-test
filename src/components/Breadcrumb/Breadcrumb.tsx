import BreadcrumbItem from '@bright/components/Breadcrumb/BreadcrumbItem'
import { BoxProps, Breadcrumb as ChakraBreadcrumb } from '@chakra-ui/core'
import React, { FC } from 'react'

type IBreadcrumbProps = Pick<BoxProps, 'children'>

interface IBreadcrumb extends FC<IBreadcrumbProps> {
  Item: typeof BreadcrumbItem
}

const Breadcrumb: IBreadcrumb = ({ children }) => {
  return <ChakraBreadcrumb>{children}</ChakraBreadcrumb>
}

Breadcrumb.Item = BreadcrumbItem

export default Breadcrumb
