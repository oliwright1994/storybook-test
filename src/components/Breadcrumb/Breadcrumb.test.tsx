import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Breadcrumb from '.'

describe('<Breadcrumb />', () => {
  const generateDefaultChildren = (): JSX.Element[] =>
    [1, 2, 3].map((value, i, arr) => (
      <Breadcrumb.Item href={`/${value}`} key={value} isCurrentPage={i === arr.length - 1}>
        {value}
      </Breadcrumb.Item>
    ))

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(<Breadcrumb>{generateDefaultChildren()}</Breadcrumb>)
    expect(container).toMatchSnapshot()
  })
})
