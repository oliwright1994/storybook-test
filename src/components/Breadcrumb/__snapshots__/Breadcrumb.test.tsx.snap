// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render as expected 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="css-0"
  >
    <ol
      class="css-0"
    >
      <li
        class="css-18biwo"
      >
        <a
          class="css-u5zpo1"
          href="/1"
        >
          1
        </a>
        <span
          class="css-1ho43uv"
          role="presentation"
        >
          /
        </span>
      </li>
      <li
        class="css-18biwo"
      >
        <a
          class="css-u5zpo1"
          href="/2"
        >
          2
        </a>
        <span
          class="css-1ho43uv"
          role="presentation"
        >
          /
        </span>
      </li>
      <li
        class="css-18biwo"
      >
        <span
          aria-current="page"
          class="css-0"
          href="/3"
        >
          3
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
